frontend/javascripts/oxalis/view/right-border-tabs/connectome_tab/connectome_view.js

```diff
@@ -0,0 +1,784 @@
+import InputComponent from "oxalis/view/components/input_component";
+import Store, {
+  type OxalisState,
+  type MutableTree,
+  type MutableNode,
+  type MutableTreeMap,
+  type ActiveMappingInfo,
+} from "oxalis/store";
+import Toast from "libs/toast";
+import getSceneController from "oxalis/controller/scene_controller_provider";
+import SynapseTree, {
+  type ConnectomeData,
+  type Agglomerate,
+  type TreeNode,
+  convertConnectomeToTreeData,
+} from "oxalis/view/right-border-tabs/connectome_tab/synapse_tree";
+import ConnectomeFilters from "oxalis/view/right-border-tabs/connectome_tab/connectome_filters";
+import ConnectomeSettings from "oxalis/view/right-border-tabs/connectome_tab/connectome_settings";
+
+const connectomeTabId = "connectome-view";
+
+type StateProps = {|
+  dataset: APIDataset,
+  // segmentationLayer will be the visible segmentation layer, or if there is none,
+  // the segmentation layer that was last visible. This is done to allow toggling
+  // the segmentation layer while browsing a connectome.
+  segmentationLayer: ?APISegmentationLayer,
+  currentConnectomeFile: ?APIConnectomeFile,
+  activeAgglomerateIds: Array<number>,
+  mappingInfo: ?ActiveMappingInfo,
+|};
+
+const mapStateToProps = (state: OxalisState): StateProps => {
+  const segmentationLayer = getVisibleOrLastSegmentationLayer(state);
+  const connectomeData =
+    segmentationLayer != null
+      ? state.localSegmentationData[segmentationLayer.name].connectomeData
+      : null;
+  const mappingInfo =
+    segmentationLayer != null
+      ? getMappingInfo(state.temporaryConfiguration.activeMappingByLayer, segmentationLayer.name)
+      : null;
+  return {
+    dataset: state.dataset,
+    segmentationLayer,
+    currentConnectomeFile: connectomeData != null ? connectomeData.currentConnectomeFile : null,
+    activeAgglomerateIds: connectomeData != null ? connectomeData.activeAgglomerateIds : [],
+    mappingInfo,
+  };
+};
+
+type Props = StateProps;
+
+type State = {
+  connectomeData: ?ConnectomeData,
+  filteredConnectomeData: ?ConnectomeData,
+  availableSynapseTypes: Array<string>,
+  checkedKeys: Array<string>,
+  expandedKeys: Array<string>,
+};
+
+const getSynapseIdsFromConnectomeData = (connectomeData: ConnectomeData): Array<number> => {
+  // Since the synapse direction filter filters the agglomerates' in/out keys and the synapse type filter
+  // filters the synapses object, the two effectively need to be unioned and only the synapse ids
+  // that occur in one of the agglomerates' in/out arrays as well as in the synapses object need to be returned.
+  const { synapses, agglomerates } = connectomeData;
+  return unique(
+    Object.values(agglomerates).flatMap(
+      // $FlowIssue[incompatible-call] remove once https://github.com/facebook/flow/issues/2221 is fixed
+      ({ in: inSynapses = [], out: outSynapses = [] }: Agglomerate) => [
+        ...inSynapses,
+        ...outSynapses,
+      ],
```

Comment:
maybe a concat is better here? not sure if this is better perf-wise (and whether it's necessary at all), but somehow the two spreads caught my eye.

```suggestion
      ({ in: inSynapses = [], out: outSynapses = [] }: Agglomerate) => inSynapses.concat(outSynapses),
```

Reply:
Seems to make a difference for large array sizes since concat is less generic and can be better optimized (https://stackoverflow.com/a/57191593). I'll switch :+1: